File: src/main/java/cn/chahuyun/economy/entity/GlobalFactor.java
package cn.chahuyun.economy.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

/**
 * 全局因子
 *
 * @author Moyuyanli
 * @date 2024/9/26 9:20
 */
@Entity(name = "GlobalFactor")
@Table
@Getter
@Setter
public class GlobalFactor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;


    /**
     * 抢劫因子<br>
     * 基础抢劫成功概率
     */
    private Double robFactor = 0.4;


    /**
     * 抢劫银行因子<br>
     * 基础抢劫成功概率
     */
    private Double robBlankFactor = 0.01;


}

File: src/main/kotlin/cn/chahuyun/economy/entity/GlobalFactor.kt
package cn.chahuyun.economy.entity

import jakarta.persistence.*

/**
 * 全局因子
 *
 * @author Moyuyanli
 * @date 2024/9/26 9:20
 */
@Entity(name = "GlobalFactor")
@Table
class GlobalFactor(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null,

    /**
     * 抢劫因子
     * 基础抢劫成功概率
     */
    var robFactor: Double = 0.4,

    /**
     * 抢劫银行因子
     * 基础抢劫成功概率
     */
    var robBlankFactor: Double = 0.01
)
